File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        left_ptr, right_ptr = 0, n - 1
        while left_ptr < right_ptr:
            curr_sum = numbers[left_ptr] + numbers[right_ptr]
            if curr_sum == target:
                return [left_ptr + 1, right_ptr + 1]
            elif curr_sum < target:
                left_ptr += 1
            else:
                right_ptr -= 1

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (seen.containsKey(nums[i])) {
                return new int[] {seen.get(nums[i]), i};
            }
            seen.put(target - nums[i], i);
        }
        return new int[] {};
    }
}
